### PR TITLE
chore!: upgrade from module template

### PR DESCRIPTION
**Pull request by waxb**
- chore: changelog updated
- chore!: upgrade module from module template and tests
- fix!: `threshold_values` to optional `amount` block to required

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -19,21 +19,20 @@ resource "google_billing_budget" "budget" {
 
   amount {
     dynamic "specified_amount" {
-      for_each = !var.use_last_period_amount ? [true] : []
+      for_each = length(try(var.amount.specified_amount, {})) != 0 ? ["specified_amount"] : []
```
Comment (waxb) [inline main.tf:22]: @mariux i am sure there is a nicer approach to this, what do you think?

**Review by zied-elouaer — APPROVED** [review]
Great job. LGTM.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,21 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 ## [Unreleased]
 
+### Added
+
+- Add module template update
+
+### Changed
+
+- Change `threshold_rules` from required to optional
+- BREAKING CHANGE: Change `amount` from optional to required and to original resource structure
```
Comment (mariux) [inline CHANGELOG.md:17]: we also changed type

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,21 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- Add module template update
+
```
Comment (mariux) [inline CHANGELOG.md:13]:
```suggestion
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,21 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+### Changed
+
+- Change `threshold_rules` from required to optional
+- BREAKING CHANGE: Change `amount` from optional to required and to original resource structure
+- Upgrade minimum supported provider version to v4.30 since it `threshold_values` isn't a required variable anymore. For details please see https://github.com/hashicorp/terraform-provider-google/blob/main/CHANGELOG.md#4300-july-25-2022
```
Comment (mariux) [inline CHANGELOG.md:18]: we also changed type and what does this mean in terms on how to upgrade?

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,21 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+### Removed
+
+- BREAKING CHANGE: Remove support for Terraform versions before v1.0
+- BREAKING CHANGE: Remove support for Terraform Google Provider versions before v4.30
```
Comment (mariux) [inline CHANGELOG.md:23]: not a breaking change as we did not remove support for last major

File: CHANGELOG.md [hunk 6]
```diff
@@ -7,6 +7,21 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+### Changed
+
+- Change `threshold_rules` from required to optional
+- BREAKING CHANGE: Change `amount` from optional to required and to original resource structure
+- Upgrade minimum supported provider version to v4.30 since it `threshold_values` isn't a required variable anymore. For details please see https://github.com/hashicorp/terraform-provider-google/blob/main/CHANGELOG.md#4300-july-25-2022
+
+### Removed
+
+- BREAKING CHANGE: Remove support for Terraform versions before v1.0
```
Comment (mariux) [inline CHANGELOG.md:22]: this wasn't supported before

File: CHANGELOG.md [hunk 7]
```diff
@@ -7,6 +7,21 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- BREAKING CHANGE: Remove support for Terraform Google Provider versions before v4.30
+
```
Comment (mariux) [inline CHANGELOG.md:24]: this hasn't changed... and new minimal version already mentioned. seems to be a copy&paste entry?

```suggestion
```

File: main.tf (hunk 1 above)
Comment (mariux) [reply main.tf:22]: please check the how to docs. this is covered in there.

File: main.tf [hunk 8]
```diff
@@ -19,21 +19,20 @@ resource "google_billing_budget" "budget" {
...
 
-    #NOTE: according to the docs, this needs to be set to null if unsed.
-    # For details please see https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/billing_budget#last_period_amount
-    last_period_amount = var.use_last_period_amount ? true : null
+    last_period_amount = try(var.amount.last_period_amount, null)
   }
 
   dynamic "threshold_rules" {
-    for_each = var.threshold_rules
+    for_each = var.threshold_rules != null ? var.threshold_rules : []
```
Comment (mariux) [inline main.tf:35]: the default should be `[]` here as also covered in the how to docs.

**Review by mariux — CHANGES_REQUESTED** [review]

**Review by zied-elouaer — DISMISSED** [review]

**Review by zied-elouaer — APPROVED** [review]

**Review by mariux — APPROVED** [review]
lgtm

**Pull request merged**